custom-images/shell_script_generator.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#            http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Shell script based image creation workflow generator.
+"""
+
+from datetime import datetime
+
+
+class Generator:
```

Comment:
This looks like maintenance nightmare :)

Is it possible to use [Python Template Engine](https://docs.python.org/3.6/library/string.html#template-strings) instead?

Reply:
Let's keep submit this version first, and switch to template in another PR.

Reply:
Done.